src/domain/session/room/timeline/linkify/linkify.js

```diff
@@ -0,0 +1,112 @@
+import { regex } from "./regex.js";
+
+export function linkify(text, callback) {
+    const matches = text.matchAll(regex);
+    let curr = 0;
+    for (let match of matches) {
+        callback(text.slice(curr, match.index), false);
```

Comment:
perhaps assign it to a variable so it's clear what is happening? Same below.
```suggestion
        const precedingText = text.slice(curr, match.index);
        callback(precedingText, false);
```